src/lib.rs

```diff
@@ -145,6 +156,39 @@ impl<'data> Iterator for DataFormatDescriptorIterator<'data> {
     }
 }
 
+struct KeyValueDataIterator<'data> {
+    data: &'data [u8],
+}
+
+impl<'data> Iterator for KeyValueDataIterator<'data> {
+    type Item = (&'data [u8], &'data [u8]);
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let mut offset = 0;
+
+        let length = bytes_to_u32(self.data, &mut offset).ok()?;
+
+        let key_and_value = &self.data[offset..offset + length as usize];
+
+        // The key is terminated with a NUL character.
+        let key_end_index = key_and_value.iter().position(|&c| c == b'\0')?;
```

Comment:
On second thought, perhaps there's an even better approach: instead of terminating the iterator early, we could interpret this case as either an empty key or an empty value. That might confuse downstream code, but it has the advantage of allowing parsing to continue on to any remaining well-formed key-value pairs.